index.js

```diff
@@ -28,230 +25,277 @@ const IMPORT_PATH_DEFAULT = {
+    }
   })
 }
 
-function getMagicWebpackComments(importArgNode) {
-  const { leadingComments } = importArgNode
-  const results = []
-  if (leadingComments && leadingComments.length) {
-    leadingComments.forEach(comment => {
-      try {
-        const validMagicString = validMagicStrings.filter(str =>
-          new RegExp(`${str}\\w*:`).test(comment.value)
-        )
-        // keep this comment if we found a match
-        if (validMagicString && validMagicString.length === 1) {
-          results.push(comment)
-        }
-      }
-      catch (e) {
-        // eat the error, but don't give up
-      }
-    })
-  }
-  return results
+function trimChunkNameBaseDir(baseDir) {
+  return baseDir.replace(/^[./]+|(\.js$)/g, '')
 }
 
-function getMagicCommentChunkName(importArgNode) {
-  const { quasis, expressions } = importArgNode
-  if (!quasis) return trimChunkNameBaseDir(importArgNode.value)
+function getMagicCommentChunkName({ expressions, quasis, value }) {
+  if (!quasis) {
+    return trimChunkNameBaseDir(value)
+  }
+
+  const [{ value: { cooked: baseDir } }] = quasis
+  const chunkName = baseDir + (expressions.length > 0 ? '[request]' : '')
 
-  const baseDir = quasis[0].value.cooked
-  const hasExpressions = expressions.length > 0
-  const chunkName = baseDir + (hasExpressions ? '[request]' : '')
   return trimChunkNameBaseDir(chunkName)
 }
 
-function getComponentId(t, importArgNode) {
-  const { quasis, expressions } = importArgNode
-  if (!quasis) return importArgNode.value
+function loadOption(p, t, template, existingChunkName) {
+  const { node, parent: IMPORT } = p
+  const chunkName = existingChunkName || getMagicCommentChunkName(node)
+  const otherValidMagicComments = getMagicWebpackComments(node)
+
+  delete node.leadingComments
+  p.addComment('leading', ` webpackChunkName: '${chunkName}' `)
+  otherValidMagicComments.forEach(validLeadingComment => p.addComment('leading', validLeadingComment.value))
+
+  const { expression: load } = template({ IMPORT })
+
+  return t.objectProperty(t.identifier('load'), load)
+}
 
-  return quasis.reduce((str, quasi, i) => {
-    const q = quasi.value.cooked
-    const id = expressions[i] && expressions[i].name
-    str += id ? `${q}\${${id}}` : q
-    return str
-  }, '')
+function getImport(p, { source, nameHint }) {
+  return addDefault(p, source, { nameHint })
 }
 
-function existingMagicCommentChunkName(importArgNode) {
-  const { leadingComments } = importArgNode
-  if (
-    leadingComments &&
-    leadingComments.length &&
-    leadingComments[0].value.indexOf('webpackChunkName') !== -1
-  ) {
+function existingMagicCommentChunkName(p) {
+  const { node: { leadingComments: [{ value: leadingComment }] = [{ value: '' }] } } = p
+  if (leadingComment.includes('webpackChunkName')) {
     try {
-      return leadingComments[0].value
+      return leadingComment
         .split('webpackChunkName:')[1]
         .replace(/["']/g, '')
         .trim()
     }
-    catch (e) {
+    catch {
```

Comment:
```suggestion
    catch (e) {
```
For better compatibility with old NodeJS versions (< 10).